app/src/test/kotlin/com/intive/picover/images/viewmodel/ImagesViewModelTest.kt

```diff
@@ -6,40 +6,37 @@ import com.intive.picover.common.viewmodel.state.ViewModelState.Error
 import com.intive.picover.common.viewmodel.state.ViewModelState.Loaded
 import com.intive.picover.common.viewmodel.state.ViewModelState.Loading
 import com.intive.picover.images.repository.ImagesRepository
-import io.kotest.core.spec.IsolationMode
 import io.kotest.core.spec.style.ShouldSpec
+import io.kotest.inspectors.forAll
 import io.kotest.matchers.shouldBe
 import io.mockk.Awaits
+import io.mockk.MockKStubScope
 import io.mockk.coEvery
 import io.mockk.just
 import io.mockk.mockk
 
 internal class ImagesViewModelTest : ShouldSpec(
 	{
 		extension(CoroutineTestExtension())
-		isolationMode = IsolationMode.InstancePerTest
 
-		val repo: ImagesRepository = mockk()
-		val tested by lazy { ImagesViewModel(repo) }
-
-		should("set Loading state WHEN initialized") {
-			coEvery { repo.fetchImages() } just Awaits
-
-			tested.state.value shouldBe Loading
-		}
-
-		should("set Error state WHEN error is thrown") {
-			val throwable = Throwable()
-			coEvery { repo.fetchImages() } throws throwable
-
-			tested.state.value shouldBe Error
-		}
-
-		should("set Success state with uris WHEN uris are returned") {
+		should("set state WHEN fetchImages called") {
 			val uris: List<Uri> = listOf(mockk())
-			coEvery { repo.fetchImages() } returns uris
-
-			tested.state.value shouldBe Loaded(uris)
+			listOf(
+				Loading to answers { just(Awaits) },
+				Error to answers { throws(Throwable()) },
+				Loaded(uris) to answers { returns(uris) },
+			).forAll { (state, answers) ->
+				val repo: ImagesRepository = mockk()
+				val tested by lazy { ImagesViewModel(repo) }
+				coEvery { repo.fetchImages() }.answers()
+
+				tested.state.value shouldBe state
+			}
 		}
 	},
 )
+
+fun answers(block: MockKStubScope<List<Uri>, List<Uri>>.() -> Unit) =
+	fun MockKStubScope<List<Uri>, List<Uri>>.() {
+		block()
+	}
```

Comment:
Really nice, I was trying to find some simplified approach but I don't have any. I think it's worth to use it anyway. Testing against repo response and view model state will be really common, so it's reasonable to invest into it and avoid repeating three boring tests in almost each view model. 
Wrap up: I would go with this solution but let's introduce some small improvements: 
1. Rename function name from `answers` to `mockkAnswer` to avoid name class with others mockk functions. 
2. Make `mockkAnswer` function to be generic to handle other types in different tests
3. Change `tested` to ordinary property, there is no need for `lazy` delegate
4. Move creation of `tested` to "when" block, since our testing part is in `init` block
5. Move `mockkAnswer` to separate file to make it available in other tests files

Later on, we can think about some e.g. testing extension to simplify parameter list (lines 25-27) but let's see 

Good job! 🍻

Reply:
I've implemented the proposed improvements